Update readme (#4)
apache/skywalking-agent-test-tool: README.md

Apache SkyWalking Agent Test Tool Suite
==========

<img src="http://skywalking.apache.org/assets/logo.svg" alt="Sky Walking logo" height="90px" align="right" />

[![Twitter Follow](https://img.shields.io/twitter/follow/asfskywalking.svg?style=for-the-badge&label=Follow&logo=twitter)](https://twitter.com/AsfSkyWalking)

![CI](https://github.com/apache/skywalking-nginx-lua/workflows/CI/badge.svg?branch=master)

[**SkyWalking**](https://github.com/apache/skywalking) Agent Test Tool is a tremendously useful test tools suite in a wide variety of languages of `Agent`. 
Includes `mock collector` and `validator`. The `mock collector` is a SkyWalking receiver, likes OAP server. 

> The `mock collector` is responsible for mocking the SkyWalking OAP receiver to collect data from the agent. It will collect data from Agent reported by `GRPC`. 

Reviewer: Remove `will`, Does this support HTTP? `it receives data from agent through GRPC and HTTP`?

And then it can be downloaded by Http API that the mock collector collected(service registered, instance registered, and segments). 

> Finally, to validate the file downloaded from `mock collector` by `SkyWalking Validator. 

Reviewer: ```suggestion
Eventually, validate the file downloaded from `mock collector` by using `SkyWalking Validator`. 
```

## Apache SkyWalking Mock Collector

- requirement:
1. JDK 1.8+
2. Maven 

- How to install

The `mock collector` is written by pure-`Java`. It compiles and packages through `Maven`.

```bash
mvn package -DskipTests
unzip ./dist/mock-collector.tar.gz -d ./mock-collector
cd ./mock-collector
bash ./bin/collector-startup.sh
```

after above steps, we can check whether the `mock collector` is available throgh HTTP API. To visit `http://localhost:12800/status` and get the reponse with `success` in the body of content.

Finally, To visit follow URL to download the data as `yaml` file, which can be used by `SkyWalking Validator Tool`.

http://localhost:12800/receiveData

- How to config on agent

> We can modify directly the conguration of agent, `./config/agent.conf`, likes following.

Reviewer 2: ```suggestion
We can modify the configurations of agent in `./config/agent.conf`, as follows
```

> ```yaml

Reviewer 2: I don't think it's yaml

# Backend service addresses.
collector.backend_service=${SW_AGENT_COLLECTOR_BACKEND_SERVICES:127.0.0.1:19876}
```

or specify the jvm arguments likes:

```bash
# Backend service addresses.
-DSW_AGENT_COLLECTOR_BACKEND_SERVICES=127.0.0.1:19876
```

## Apache SkyWalking Validator Tool

> `Validator Tool` a data validating tool. It resposible for validating the `expected data` with `actual data`.

Reviewer 2: ```suggestion
`Validator Tool` is a data validating tool. It is responsible for validating the `expected data` with `actual data`.
```

```bash
java -jar \
    -Xmx256m -Xms256m \

>     -DcaseName="${SCENARIO_NAME}-${SCENARIO_VERSION}" \

Reviewer: This parameter value demonstration is not clear, should be `path/to/expected-file-folder`?

>     -DtestCasePath=${SCENARIO_HOME}/data/ \

Reviewer: `path/to/download-folder` ?
